Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Text;
+
+namespace Polynomials.Core
+{
+    public class Polynomial : IEquatable<Polynomial>, ICloneable
+    {
+        private readonly float[] _coefficients;
+
+        public float this[int i]
+        {
+            get => _coefficients[i];
+            set => _coefficients[i] = value;
+        }
+
+        public Polynomial(float[] coefficients)
+        {
+            if (coefficients.Length > 100)
+            {
+                throw new Exception("Max power must be less or equal 100");
+            }
+
+            _coefficients = coefficients;
+        }
+
+        public bool Equals(Polynomial? other)
+        {
+            if (this is null || other is null)
+            {
+                return false;
+            }
+
+            if (this._coefficients.Length != other._coefficients.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < this._coefficients.Length; i++)
+            {
+                if (!FloatsAreEqualEnough(this[i], other[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        // This function made public to be able to test in xUnit
+        public static bool FloatsAreEqualEnough(float left, float right)
+        {
+            const float tolerance = 0.000000001F;
+            //return Math.Abs(left - right) < tolerance;
+            if (Math.Abs(left - right) <= tolerance)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public static Polynomial operator +(Polynomial? left, Polynomial? right)
+        {
+            if (left is null || !(right is null))
+            {
+                var polynomial = right.Clone();
+                return (Polynomial)polynomial;
+            }
+
+            if (!(left is null) || right is null)
+            {
+                var polynomial = left.Clone();
```

Comment:
You can shorten that by just 
`return (Polynomial) left.Clone();`

Same above